client/lib/purchases/index.js

```diff
@@ -85,14 +85,21 @@ function getPartnerName( purchase ) {
 	return null;
 }
 
+// TODO: refactor to avoid returning a localized string.
 function getSubscriptionEndDate( purchase ) {
-	return purchase.expiryMoment.format( 'LL' );
+	const localeSlug = i18n.getLocaleSlug();
+	return (
+		purchase.expiryDateValid &&
+		moment( purchase.expiryDate, purchase.expiryDateFormat )
+			.locale( localeSlug )
+			.format( 'LL' )
+	);
```

Comment:
If the expiry date is not valid, then this function returns `false` and we'll get a UI message like:
```
Your purchase is available until it expires on false
```
The original code would probably just display `null` instead of `false`, though.

Are there really any purchases that never expire?

Reply:
I'm not sure if you can have purchases that don't expire. I've changed the code here, in any case.